scalecube-organization/src/main/java/io/scalecube/organization/repository/couchbase/CouchbaseEntityRepository.java

```diff
@@ -31,34 +30,24 @@
 
   private static final String BUCKET_PASSWORD = ".bucket.password";
   private static final String BUCKET = ".bucket";
-  private static final Object lock = new Object();
-  private final CouchbaseSettings settings;
-  private final Class<T> type;
   private final CouchbaseExceptionTranslator exceptionTranslator =
       new CouchbaseExceptionTranslator();
-  private CouchbaseCluster cluster;
-  private TranslationService translationService = new JacksonTranslationService();
+  private final TranslationService translationService = new JacksonTranslationService();
+  protected final CouchbaseSettings settings;
+  protected final Cluster cluster;
+  private final Class<T> type;
   private String bucketName;
-  private String couchbaseUsername;
-  private String couchbasePassword;
-
-  CouchbaseEntityRepository(String alias, Class<T> type) {
-    requireNonNull(type);
-
-    this.type = type;
-    this.settings = new CouchbaseSettings.Builder().build();
 
+  CouchbaseEntityRepository(
+      CouchbaseSettings settings, Cluster cluster, String alias, Class<T> type) {
+    this.settings = requireNonNull(settings);
+    this.cluster = requireNonNull(cluster);
+    this.type = requireNonNull(type);
     if (alias != null) {
-      this.bucketName = getBucketName(alias);
-      this.couchbaseUsername = settings.getCouchbaseUsername();
-      this.couchbasePassword = settings.getCouchbasePassword();
+      this.bucketName = settings.getProperty(alias + BUCKET);

```

Comment:
Hard to believe that `bucketName` can be null and app would still operate correctly   .. o_O   . Let's fail fast and don't allow to not to have bucketName. I.e. if `alias` null -> throw , if prop from settings null -> throw.

Reply:
I can' do it.